### PR TITLE
qcustomplot: Update dependencies and add package type

### PR DESCRIPTION
**Pull request by jwillikers**
Specify library name and version:  **qcustomplot/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by RubenRBS** [comment]
Seems like you're missing a `transitive_headers=True` trait in the requires of qt, it compiles locally for me after that and makes sense given the code.

Maybe you also need `transitive_libs`, that one I haven't taken the time to check

**Comment by RubenRBS** [comment]
Launched build for missing zstd

File: recipes/qcustomplot/all/conanfile.py [hunk 1]
```diff
@@ -45,9 +46,9 @@ def configure(self):
 
     def requirements(self):
         if Version(self.version) >= "2.0.0":
-            self.requires("qt/6.4.1")
+            self.requires("qt/6.6.0", transitive_headers=True)
```
Comment (valgur) [inline recipes/qcustomplot/all/conanfile.py:49]:
```suggestion
            self.requires("qt/6.6.1", transitive_headers=True, transitive_libs=True)
```
The missing binary issue for Qt6 should be resolved for now, I think.

File: recipes/qcustomplot/all/conanfile.py [hunk 2]
```diff
@@ -45,9 +46,9 @@ def configure(self):
...
         else:
-            self.requires("qt/5.15.7")
+            self.requires("qt/5.15.11", transitive_headers=True)
```
Comment (valgur) [inline recipes/qcustomplot/all/conanfile.py:51]:
```suggestion
            self.requires("qt/5.15.11", transitive_headers=True, transitive_libs=True)
```
Qt probably isn't used as header-only here, so transitive_libs is likely necessary.

**Comment by valgur** [comment]
It probably requires a `tool_requires` on Qt and a VirtualBuildEnv + VirtualRunEnv, just like in #23645.

Isn't this PR mostly superseded by #23645 currently?

**Comment by jwillikers** [comment]
> It probably requires a `tool_requires` on Qt and a VirtualBuildEnv + VirtualRunEnv, just like in #23645.
> 
> Isn't this PR mostly superseded by #23645 currently?

I wasn't aware of that one #23645. I can close in favor of that.

**Pull request closed without merging**